script.js

```diff
@@ -0,0 +1,333 @@
+/* global SpatialNavigator */
+
+window.SmartScrollbox = (function(exports) {
+  'use strict';
+
+  var proto = Object.create(HTMLElement.prototype);
+
+  proto.createdCallback = function ssb_createdCallback() {
+
+    this._template = template.content.cloneNode(true);
+
+    this.listElem = this._template.getElementById('scrollbox-list');
+    this.createShadowRoot().appendChild(this._template);
+
+    // left and right margin of the scroll list
+    this.margin = this.getAttribute('margin') || 0;
+    this.margin = parseInt(this.margin, 10) * 10;
+
+    this.viewWidth = parseInt(this.clientWidth - 2 * this.margin, 10);
+
+    this.translateX = 0;
+
+    // Current active node
+    this.currentNode = null;
+
+    // add navigable elements to spacial navigator
+    var navigableElems = Array.prototype.slice.call(
+                                  this.getElementsByClassName('navigable'));
+    if (navigableElems.length > 0) {
+      this._scrollTo(navigableElems[0]);
+    }
+    this._spatialNavigator = new SpatialNavigator(navigableElems);
+    this._spatialNavigator.on('focus', this._handleFocus.bind(this));
+    this._spatialNavigator.on('unfocus', this._handleUnfocus.bind(this));
+
+    // handle appendChild and removeChild in scrollbox
+    this._observer = new MutationObserver(this._handleMutation.bind(this));
+    this._observer.observe(this, {
+      childList: true
+    });
+
+    this.addEventListener('transitionend', this);
+    this.addEventListener('scroll', this);
+    this.addEventListener('focus', this);
+    this.addEventListener('blur', this);
+  };
+
+  /**
+   * Current focused element
+   */
+  Object.defineProperty(proto, 'currentElement', {
+    get: function() {
+      return this._spatialNavigator.getFocusedElement();
+    }
+  });
+
+  /**
+   * Blur the currentElement
+   */
+  proto.blur = function() {
+    if (this.currentElement) {
+      this.currentElement.blur();
+    }
+  };
+
+  /**
+   * Focus a element in scrollbox. If the input item is a number, it will focus
+   * the first navigable element in node having the same index as the number.
```

Comment:
'the first navigable element having the same index' sounds weird, isn't index unique?